File: 0441-arranging-coins/0441-arranging-coins.java
class Solution {
    public int arrangeCoins(int n) {
        // i = rows of coins
        int i = 1;
        while (n >= i){
            // n-i gives the coins to be placed
            n -= i;
            // increment the rows
            i += 1;
        }
// return the rows upto which the coins are placed (i-1) because if the coins are not completely placed
        return i-1;
    }
}

File: 0441-arranging-coins/0441-arranging-coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
# i= number of coins placed
        i = 1
# arranging the coins from 1 
        while n >= i:
            n -= i
            i += 1
        return i - 1
# while number of coins is greater than the number of coins placed
# decrement n, i.e n will be remaining coins
# increment i, i.e wiil be the rows placed